source/services/relay.md

```diff
@@ -67,13 +68,14 @@ More info about the signature values in appendix F of the [Ethereum Yellow Paper
 **Note:** Atomic operation, many values of s are invalid which are generated by the server.
 
 Clients should verify the server’s response with the following process:
-1. Reconstruct sender address from `signature`
-1. Reconstruct Safe address with sender address and `nonce = 0`
-1. Verify that `Safe` checksum and reconstructed Safe addresses checksum are matching
-1. Verify that `signature` is correct for hash of `tx`
-1. Verify that `tx` has valid bytecode (postponed until mainnet release).
-1. If all checks pass, then transaction and Safe address are valid and user can transfer at least `payment` amount of ETH to the Safe address.
-1. Otherwise, the response has error or it is compromised, and it should not be used any further.
+1. Verify that the `s` that you provided in the request matches the `s` returned in the response.
+1. Verify that `tx.data` matches the bytecode of the Gnosis Safe Proxy contract with the correct owners and threshold. The `value` and `nonce` of the transaction should be zero.
```

Comment:
I still think we should say a little more here (on how to check). But maybe that is for later.

But we should at least mention that they should also check `mastercopy` (since we deploy a proxy), `funder` (account that will receive the payment), `payment` and `paymentToken`.

We could probably link the `PayingProxy.sol` and `GnosisSafe.setup()` to provide info what should be checked. And also link the release page of the contract as the source for the master copy addresses (and maybe also the API `about` endpoint)

Reply:
Agreed. Maybe we can merge this one since it should be also correct and add more details later on.

Reply:
Since we will probably not do that soonish (at least as far as I know us), I would say lets add the basic stuff now. As said checking the constructor parameter for the Proxy is very important (as important as the setup parameter)